src/gt4py/next/otf/compilation/build_systems/cmake_lists.py

```diff
@@ -36,16 +44,20 @@ class CMakeListsFile(eve.Node):
     link_deps: Sequence[LinkDependency]
     source_names: Sequence[str]
     bin_output_suffix: str
+    extra_languages: Sequence[ExtraLanguage]
 
 
 class CMakeListsGenerator(eve.codegen.TemplatedGenerator):
+    # TODO(ricoh): enable_language(CUDA) has to be switchable
     CMakeListsFile = as_jinja(
-        """
+        textwrap.dedent(
+            """
         project({{project_name}})
         cmake_minimum_required(VERSION 3.20.0)
 
         # Languages
         enable_language(CXX)
+        {{"\\n".join(extra_languages)}}
```

Comment:
in a discussion with @egparedes we realized that probably we can just do 

```
include(CheckLanguage)
check_language(CUDA)
if(CMAKE_CUDA_COMPILER)
  enable_language(CUDA)
endif()
```
Ideally we would make the result of the check available to the Python side, because we can then error nicely, but as a first step we could just error by forwarding the CMake error which probably would be something like "`gtfn_gpu` is not a target".